src/compiler/emitter.ts

```diff
@@ -725,7 +725,8 @@ namespace ts {
             fileExists: f => host.fileExists(f),
             directoryExists: host.directoryExists && (f => host.directoryExists!(f)),
             useCaseSensitiveFileNames: () => host.useCaseSensitiveFileNames(),
-            getProgramBuildInfo: returnUndefined
+            getProgramBuildInfo: returnUndefined,
+            getSourceFileFromReference: returnUndefined, // TODO: Implement?

```

Comment:
Should this be `notImplemented`. As this is just writing existing texts by updating prepend outputs.

Reply:
`notImplemented` causes a crash, since the method is actually used in the declaration emitter. Previously it would have relied on `getSourceFile` which is a `returnUndefined`, so using `returnUndefined` means the behavior hasn't changed.

Reply:
I see. Then it should be as you have `returnUndefined` since this emit is not going to recalculate the refs. Also since `getSourceFile` and `getSourceFileByPath` are `returnUndefined` as well.